src/Resources/ResourceManager/Implementation/Deployments/SaveAzureDeploymentTemplateCmdlet.cs

```diff
@@ -87,5 +109,65 @@ protected override void OnProcessRecord()
                 WriteObject(new PSTemplatePath() { Path = path });
             }
         }
+
+        private string GetDeploymentTemplate(bool isTenantDeployment, string managementGroupId, string resourceGroupName, string deploymentName)
+        {
+            if (isTenantDeployment)
+            {
+                return ResourceManagerSdkClient.GetDeploymentTemplateAtTenantScope(deploymentName);
+            }
+            else if (!string.IsNullOrEmpty(managementGroupId))
+            {
+                return ResourceManagerSdkClient.GetDeploymentTemplateAtManagementGroup(managementGroupId, deploymentName);
+            }
+            else if (!string.IsNullOrEmpty(resourceGroupName))
+            {
+                return ResourceManagerSdkClient.GetDeploymentTemplateAtResourceGroup(resourceGroupName, deploymentName);
+            }
+            else
+            {
+                return ResourceManagerSdkClient.GetDeploymentTemplateAtSubscrpitionScope(deploymentName);
+            }
+        }
+
+        private string GetDeploymetName()
+        {
+            if (!string.IsNullOrEmpty(this.DeploymentName))
+            {
+                return this.DeploymentName;
+            }
+            else
+            {
+                return this.DeploymentObject.DeploymentName;
+            }
+        }
+
+        private string GetManagementGroupId()
+        {
+            if (!string.IsNullOrEmpty(this.ManagementGroupId))
+            {
+                return this.ManagementGroupId;
+            }
+            else if (this.DeploymentObject != null)
+            {
+                return this.DeploymentObject.ManagementGroupId;
+            }
+
+            return null;
+        }
+
+        private string GetResourceGroupName()
+        {
+            if (!string.IsNullOrEmpty(this.ResourceGroupName))
+            {
+                return this.ResourceGroupName;
+            }
+            else if (this.DeploymentObject != null)
+            {
+                return this.DeploymentObject.ResourceGroupName;
+            }
+
+            return null;
+        }
```

Comment:
Are some of these repeated? Should we throw them into some base class?

Reply:
sure. will take a look.